[testing] test_trainer.py is failing

```
pytest tests/test_trainer.py
```
```
platform linux -- Python 3.7.9, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /mnt/nvme1/code/huggingface/transformers-master
plugins: xdist-2.1.0, forked-1.3.0
collected 11 items                                                                                                                                   

tests/test_trainer.py F.FF.FF...F                                                                                                              [100%]

====================================================================== FAILURES ======================================================================
____________________________________________________ TrainerIntegrationTest.test_custom_optimizer ____________________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_custom_optimizer>

    def test_custom_optimizer(self):
        train_dataset = RegressionDataset()
        args = TrainingArguments("./regression")
        model = RegressionModel()
        optimizer = torch.optim.SGD(model.parameters(), lr=1.0)
        lr_scheduler = torch.optim.lr_scheduler.LambdaLR(optimizer, lr_lambda=lambda x: 1.0)
        trainer = Trainer(model, args, train_dataset=train_dataset, optimizers=(optimizer, lr_scheduler))
        trainer.train()
    
>       self.assertTrue(torch.abs(trainer.model.a - 1.8950) < 1e-4)
E       AssertionError: tensor(False, device='cuda:0') is not true

tests/test_trainer.py:240: AssertionError
---------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------
Iteration: 100%|██████████| 4/4 [00:00<00:00,  4.15it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 584.41it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 570.73it/s]
Epoch: 100%|██████████| 3/3 [00:00<00:00,  3.06it/s]
_______________________________________________________ TrainerIntegrationTest.test_model_init _______________________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_model_init>

    def test_model_init(self):
        train_dataset = RegressionDataset()
        args = TrainingArguments("./regression", learning_rate=0.1)
        trainer = Trainer(args=args, train_dataset=train_dataset, model_init=lambda: RegressionModel())
        trainer.train()
>       self.check_trained_model(trainer.model)

tests/test_trainer.py:249: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_trainer.py:105: in check_trained_model
    self.assertTrue(torch.abs(model.a - 0.6975) < 1e-4)
E   AssertionError: tensor(False, device='cuda:0') is not true
---------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------
Iteration: 100%|██████████| 4/4 [00:00<00:00, 540.05it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 510.99it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 553.37it/s]
Epoch: 100%|██████████| 3/3 [00:00<00:00, 130.01it/s]
______________________________________________ TrainerIntegrationTest.test_number_of_steps_in_training _______________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_number_of_steps_in_training>

    def test_number_of_steps_in_training(self):
        # Regular training has n_epochs * len(train_dl) steps
        trainer = get_regression_trainer(learning_rate=0.1)
        train_output = trainer.train()
>       self.assertEqual(train_output.global_step, self.n_epochs * 64 / self.batch_size)
E       AssertionError: 12 != 24.0

tests/test_trainer.py:129: AssertionError
---------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------
Iteration: 100%|██████████| 4/4 [00:00<00:00, 547.43it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 573.03it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 557.12it/s]
Epoch: 100%|██████████| 3/3 [00:00<00:00, 136.18it/s]
_________________________________________________ TrainerIntegrationTest.test_reproducible_training __________________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_reproducible_training>

    def test_reproducible_training(self):
        # Checks that training worked, model trained and seed made a reproducible training.
        trainer = get_regression_trainer(learning_rate=0.1)
        trainer.train()
>       self.check_trained_model(trainer.model)

tests/test_trainer.py:118: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_trainer.py:105: in check_trained_model
    self.assertTrue(torch.abs(model.a - 0.6975) < 1e-4)
E   AssertionError: tensor(False, device='cuda:0') is not true
---------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------
Iteration: 100%|██████████| 4/4 [00:00<00:00, 388.21it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 556.31it/s]
Iteration: 100%|██████████| 4/4 [00:00<00:00, 544.31it/s]
Epoch: 100%|██████████| 3/3 [00:00<00:00, 117.50it/s]
_______________________________________________ TrainerIntegrationTest.test_train_and_eval_dataloaders _______________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_train_and_eval_dataloaders>

    def test_train_and_eval_dataloaders(self):
        trainer = get_regression_trainer(learning_rate=0.1, per_device_train_batch_size=16)
>       self.assertEqual(trainer.get_train_dataloader().batch_size, 16)
E       AssertionError: 32 != 16

tests/test_trainer.py:143: AssertionError
____________________________________________________ TrainerIntegrationTest.test_trainer_with_nlp ____________________________________________________

self = <tests.test_trainer.TrainerIntegrationTest testMethod=test_trainer_with_nlp>

    def test_trainer_with_nlp(self):
        np.random.seed(42)
        x = np.random.normal(size=(64,)).astype(np.float32)
        y = 2.0 * x + 3.0 + np.random.normal(scale=0.1, size=(64,))
        train_dataset = nlp.Dataset.from_dict({"input_x": x, "label": y})
    
        # Base training. Should have the same results as test_reproducible_training
        model = RegressionModel()
        args = TrainingArguments("./regression", learning_rate=0.1)
>       trainer = Trainer(model, args, train_dataset=train_dataset)

tests/test_trainer.py:212: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/transformers/trainer.py:285: in __init__
    self._remove_unused_columns(self.train_dataset, description="training")
src/transformers/trainer.py:311: in _remove_unused_columns
    dataset.set_format(type=dataset.format["type"], columns=columns)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Dataset(features: {'input_x': Value(dtype='float32', id=None), 'label': Value(dtype='float64', id=None)}, num_rows: 64), type = 'python'
columns = ['input_x', 'label'], output_all_columns = False, format_kwargs = {}

    def set_format(
        self,
        type: Optional[str] = None,
        columns: Optional[List] = None,
        output_all_columns: bool = False,
        **format_kwargs,
    ):
        """ Set __getitem__ return format (type and columns)
    
            Args:
                type (Optional ``str``): output type selected in [None, 'numpy', 'torch', 'tensorflow', 'pandas']
                    None means __getitem__ returns python objects (default)
                columns (Optional ``List[str]``): columns to format in the output
                    None means __getitem__ returns all columns (default)
                output_all_columns (``bool`` default to False): keep un-formated columns as well in the output (as python objects)
                format_kwargs: keywords arguments passed to the convert function like `np.array`, `torch.tensor` or `tensorflow.ragged.constant`.
        """
        # Check return type
        if type == "torch":
            try:
                import torch  # noqa: F401
            except ImportError:
                logger.error("PyTorch needs to be installed to be able to return PyTorch tensors.")
        elif type == "tensorflow":
            try:
                import tensorflow  # noqa: F401
            except ImportError:
                logger.error("Tensorflow needs to be installed to be able to return Tensorflow tensors.")
        else:
            assert not (
                type == "pandas" and (output_all_columns or format_kwargs)
            ), "Format type 'pandas' doesn't allow the use of `output_all_columns` or `**format_kwargs`."
            assert (
                type is None or type == "numpy" or type == "pandas"
>           ), "Return type should be None or selected in ['numpy', 'torch', 'tensorflow', 'pandas']."
E           AssertionError: Return type should be None or selected in ['numpy', 'torch', 'tensorflow', 'pandas'].

/home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/nlp/arrow_dataset.py:542: AssertionError
================================================================== warnings summary ==================================================================
/home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/data/ops/iterator_ops.py:546
  /home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/data/ops/iterator_ops.py:546: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    class IteratorBase(collections.Iterator, trackable.Trackable,

/home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/data/ops/dataset_ops.py:106
  /home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/data/ops/dataset_ops.py:106: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    class DatasetV2(collections.Iterable, tracking_base.Trackable,

/home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/autograph/utils/testing.py:21
  /home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/tensorflow/python/autograph/utils/testing.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

src/transformers/modeling_tf_utils.py:718
  /mnt/nvme1/code/huggingface/transformers-master/src/transformers/modeling_tf_utils.py:718: DeprecationWarning: invalid escape sequence \s
    """

src/transformers/modeling_funnel.py:130
  /mnt/nvme1/code/huggingface/transformers-master/src/transformers/modeling_funnel.py:130: DeprecationWarning: invalid escape sequence \d
    layer_index = int(re.search("layer_(\d+)", m_name).groups()[0])

tests/test_trainer.py::TrainerIntegrationTest::test_custom_optimizer
tests/test_trainer.py::TrainerIntegrationTest::test_evaluate
tests/test_trainer.py::TrainerIntegrationTest::test_model_init
tests/test_trainer.py::TrainerIntegrationTest::test_number_of_steps_in_training
tests/test_trainer.py::TrainerIntegrationTest::test_predict
tests/test_trainer.py::TrainerIntegrationTest::test_reproducible_training
  /home/stas/anaconda3/envs/main-37/lib/python3.7/site-packages/torch/nn/parallel/_functions.py:61: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
    warnings.warn('Was asked to gather along dimension 0, but all '

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================== short test summary info ===============================================================
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_custom_optimizer - AssertionError: tensor(False, device='cuda:0') is not true
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_model_init - AssertionError: tensor(False, device='cuda:0') is not true
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_number_of_steps_in_training - AssertionError: 12 != 24.0
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_reproducible_training - AssertionError: tensor(False, device='cuda:0') is not true
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_train_and_eval_dataloaders - AssertionError: 32 != 16
FAILED tests/test_trainer.py::TrainerIntegrationTest::test_trainer_with_nlp - AssertionError: Return type should be None or selected in ['numpy', '...
===================================================== 6 failed, 5 passed, 11 warnings in 10.28s ==================================================
```
Env:
```
- `transformers` version: 3.1.0
- Platform: Linux-4.15.0-112-generic-x86_64-with-debian-buster-sid
- Python version: 3.7.9
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): 2.3.0 (True)
- Using GPU in script?: yes
```

Thanks.

 @sgugger?

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -77,7 +77,7 @@ jobs:
                       - v0.3-torch_and_tf-{{ checksum "setup.py" }}
                       - v0.3-{{ checksum "setup.py" }}
             - run: pip install --upgrade pip
-            - run: pip install git+https://github.com/huggingface/nlp
+            - run: pip install git+https://github.com/huggingface/datasets
             - run: pip install .[sklearn,tf-cpu,torch,testing]
             - run: pip install codecov pytest-cov
             - save_cache:
@@ -104,7 +104,7 @@ jobs:
                       - v0.3-torch-{{ checksum "setup.py" }}
                       - v0.3-{{ checksum "setup.py" }}
             - run: pip install --upgrade pip
-            - run: pip install git+https://github.com/huggingface/nlp
+            - run: pip install git+https://github.com/huggingface/datasets
             - run: pip install .[sklearn,torch,testing]
             - save_cache:
                   key: v0.3-torch-{{ checksum "setup.py" }}
@@ -129,7 +129,7 @@ jobs:
                       - v0.3-tf-{{ checksum "setup.py" }}
                       - v0.3-{{ checksum "setup.py" }}
             - run: pip install --upgrade pip
-            - run: pip install git+https://github.com/huggingface/nlp
+            - run: pip install git+https://github.com/huggingface/datasets
             - run: pip install .[sklearn,tf-cpu,testing]
             - save_cache:
                   key: v0.3-tf-{{ checksum "setup.py" }}

--- a/.github/workflows/self-push.yml
+++ b/.github/workflows/self-push.yml
@@ -46,7 +46,7 @@ jobs:
         pip install --upgrade pip
         pip install torch!=1.6.0
         pip install .[sklearn,testing,onnxruntime]
-        pip install git+https://github.com/huggingface/nlp
+        pip install git+https://github.com/huggingface/datasets
 
     - name: Are GPUs recognized by our DL frameworks
       run: |

--- a/.github/workflows/self-scheduled.yml
+++ b/.github/workflows/self-scheduled.yml
@@ -43,7 +43,7 @@ jobs:
         pip install --upgrade pip
         pip install torch!=1.6.0
         pip install .[sklearn,testing,onnxruntime]
-        pip install git+https://github.com/huggingface/nlp
+        pip install git+https://github.com/huggingface/datasets
 
     - name: Are GPUs recognized by our DL frameworks
       run: |

--- a/examples/longform-qa/README.md
+++ b/examples/longform-qa/README.md
@@ -1,5 +1,5 @@
 # Long Form Question Answering
 
-This folder contains the code for the Long Form Question answering [demo](http://35.226.96.115:8080/) as well as methods to train and use a fully end-to-end Long Form Question Answering system using the [🤗transformers](https://github.com/huggingface/transformers) and [🤗nlp](https://github.com/huggingface/nlp) libraries.
+This folder contains the code for the Long Form Question answering [demo](http://35.226.96.115:8080/) as well as methods to train and use a fully end-to-end Long Form Question Answering system using the [🤗transformers](https://github.com/huggingface/transformers) and [🤗datasets](https://github.com/huggingface/datasets) libraries.
 
 You can use these methods to train your own system by following along the associate [notebook](https://github.com/huggingface/notebooks/blob/master/longform-qa/Long_Form_Question_Answering_with_ELI5_and_Wikipedia.ipynb) or [blog post](https://yjernite.github.io/lfqa.html).

--- a/examples/longform-qa/eli5_app.py
+++ b/examples/longform-qa/eli5_app.py
@@ -1,11 +1,10 @@
+import datasets
 import faiss
-import nlp
 import numpy as np
 import streamlit as st
 import torch
 from elasticsearch import Elasticsearch
 
-import transformers
 from eli5_utils import (
     embed_questions_for_retrieval,
     make_qa_s2s_model,
@@ -45,7 +44,7 @@ def load_models():
 def load_indexes():
     if LOAD_DENSE_INDEX:
         faiss_res = faiss.StandardGpuResources()
-        wiki40b_passages = nlp.load_dataset(path="wiki_snippets", name="wiki40b_en_100_0")["train"]
+        wiki40b_passages = datasets.load_dataset(path="wiki_snippets", name="wiki40b_en_100_0")["train"]
         wiki40b_passage_reps = np.memmap(
             "wiki40b_passages_reps_32_l-8_h-768_b-512-512.dat",
             dtype="float32",
@@ -63,7 +62,7 @@ def load_indexes():
 
 @st.cache(allow_output_mutation=True)
 def load_train_data():
-    eli5 = nlp.load_dataset("eli5", name="LFQA_reddit")
+    eli5 = datasets.load_dataset("eli5", name="LFQA_reddit")
     eli5_train = eli5["train_eli5"]
     eli5_train_q_reps = np.memmap(
         "eli5_questions_reps.dat", dtype="float32", mode="r", shape=(eli5_train.num_rows, 128)

--- a/examples/longform-qa/eli5_app.py
+++ b/examples/longform-qa/eli5_app.py
@@ -5,6 +5,7 @@ import streamlit as st
 import torch
 from elasticsearch import Elasticsearch
 
+import transformers
 from eli5_utils import (
     embed_questions_for_retrieval,
     make_qa_s2s_model,

--- a/examples/longform-qa/eli5_utils.py
+++ b/examples/longform-qa/eli5_utils.py
@@ -4,8 +4,8 @@ import os  # noqa: F401
 from random import choice, randint
 from time import time
 
+import datasets  # noqa: F401
 import faiss  # noqa: F401
-import nlp  # noqa: F401
 import numpy as np
 import pandas as pd
 import torch

--- a/examples/requirements.txt
+++ b/examples/requirements.txt
@@ -12,7 +12,7 @@ faiss
 streamlit
 elasticsearch
 pandas
-nlp
+datasets
 fire
 pytest
 conllu

--- a/examples/seq2seq/download_wmt.py
+++ b/examples/seq2seq/download_wmt.py
@@ -5,25 +5,25 @@ from tqdm import tqdm
 
 
 def download_wmt_dataset(src_lang="ro", tgt_lang="en", dataset="wmt16", save_dir=None) -> None:
-    """Download a dataset using the nlp package and save it to the format expected by finetune.py
+    """Download a dataset using the datasets package and save it to the format expected by finetune.py
     Format of save_dir: train.source, train.target, val.source, val.target, test.source, test.target.
 
     Args:
         src_lang: <str> source language
         tgt_lang: <str> target language
-        dataset: <str> wmt16, wmt17, etc. wmt16 is a good start as it's small. To get the full list run `import nlp; print([d.id for d in nlp.list_datasets() if "wmt" in d.id])`
+        dataset: <str> wmt16, wmt17, etc. wmt16 is a good start as it's small. To get the full list run `import datasets; print([d.id for d in datasets.list_datasets() if "wmt" in d.id])`
         save_dir: <str>, where to save the datasets, defaults to f'{dataset}-{src_lang}-{tgt_lang}'
 
     Usage:
         >>> download_wmt_dataset('ro', 'en', dataset='wmt16') # saves to wmt16-ro-en
     """
     try:
-        import nlp
+        import datasets
     except (ModuleNotFoundError, ImportError):
-        raise ImportError("run pip install nlp")
+        raise ImportError("run pip install datasets")
     pair = f"{src_lang}-{tgt_lang}"
     print(f"Converting {dataset}-{pair}")
-    ds = nlp.load_dataset(dataset, pair)
+    ds = datasets.load_dataset(dataset, pair)
     if save_dir is None:
         save_dir = f"{dataset}-{pair}"
     save_dir = Path(save_dir)

--- a/setup.cfg
+++ b/setup.cfg
@@ -7,6 +7,7 @@ known_first_party = transformers
 known_third_party =
     absl
     conllu
+    datasets
     elasticsearch
     fairseq
     faiss
@@ -16,7 +17,6 @@ known_third_party =
     git
     h5py
     matplotlib
-    nlp
     nltk
     numpy
     packaging

--- a/src/transformers/__init__.py
+++ b/src/transformers/__init__.py
@@ -83,7 +83,7 @@ from .file_utils import (
     add_start_docstrings,
     cached_path,
     is_apex_available,
-    is_nlp_available,
+    is_datasets_available,
     is_psutil_available,
     is_py3nvml_available,
     is_tf_available,

--- a/src/transformers/file_utils.py
+++ b/src/transformers/file_utils.py
@@ -66,12 +66,12 @@ except (ImportError, AssertionError):
 
 
 try:
-    import nlp  # noqa: F401
+    import datasets  # noqa: F401
 
-    _nlp_available = True
+    _datasets_available = True
 
 except ImportError:
-    _nlp_available = False
+    _datasets_available = False
 
 try:
     from torch.hub import _get_torch_home
@@ -155,8 +155,8 @@ def is_torch_tpu_available():
     return _torch_tpu_available
 
 
-def is_nlp_available():
-    return _nlp_available
+def is_datasets_available():
+    return _datasets_available
 
 
 def is_psutil_available():

--- a/src/transformers/trainer.py
+++ b/src/transformers/trainer.py
@@ -20,7 +20,7 @@ from torch.utils.data.sampler import RandomSampler, Sampler, SequentialSampler
 from tqdm.auto import tqdm, trange
 
 from .data.data_collator import DataCollator, DataCollatorWithPadding, default_data_collator
-from .file_utils import is_nlp_available, is_torch_tpu_available
+from .file_utils import is_datasets_available, is_torch_tpu_available
 from .integrations import (
     default_hp_search_backend,
     is_comet_available,
@@ -65,8 +65,8 @@ else:
     _use_native_amp = True
     from torch.cuda.amp import autocast
 
-if is_nlp_available():
-    import nlp
+if is_datasets_available():
+    import datasets
 
 if is_torch_tpu_available():
     import torch_xla.core.xla_model as xm
@@ -179,10 +179,10 @@ class Trainer:
             :obj:`eval_dataset`. Will default to :func:`~transformers.default_data_collator` if no ``tokenizer`` is
             provided, an instance of :func:`~transformers.DataCollatorWithPadding` otherwise.
         train_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
-            The dataset to use for training. If it is an :obj:`nlp.Dataset`, columns not accepted by the
+            The dataset to use for training. If it is an :obj:`datasets.Dataset`, columns not accepted by the
             ``model.forward()`` method are automatically removed.
         eval_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
-             The dataset to use for evaluation. If it is an :obj:`nlp.Dataset`, columns not accepted by the
+             The dataset to use for evaluation. If it is an :obj:`datasets.Dataset`, columns not accepted by the
             ``model.forward()`` method are automatically removed.
         tokenizer (:class:`PreTrainedTokenizerBase`, `optional`):
             The tokenizer used to preprocess the data. If provided, will be used to automatically pad the inputs the
@@ -280,10 +280,10 @@ class Trainer:
                 FutureWarning,
             )
 
-        if is_nlp_available():
-            if isinstance(train_dataset, nlp.Dataset):
+        if is_datasets_available():
+            if isinstance(train_dataset, datasets.Dataset):
                 self._remove_unused_columns(self.train_dataset, description="training")
-            if isinstance(eval_dataset, nlp.Dataset):
+            if isinstance(eval_dataset, datasets.Dataset):
                 self._remove_unused_columns(self.eval_dataset, description="evaluation")
 
         self.global_step = None
@@ -294,7 +294,7 @@ class Trainer:
         self.hp_search_backend = None
         self.use_tune_checkpoints = False
 
-    def _remove_unused_columns(self, dataset: "nlp.Dataset", description: Optional[str] = None):
+    def _remove_unused_columns(self, dataset: "datasets.Dataset", description: Optional[str] = None):
         if not self.args.remove_unused_columns:
             return
         # Inspect model forward signature to keep only the arguments it accepts.
@@ -364,12 +364,12 @@ class Trainer:
 
         Args:
             eval_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
-                If provided, will override :obj:`self.eval_dataset`. If it is an :obj:`nlp.Dataset`, columns not
+                If provided, will override :obj:`self.eval_dataset`. If it is an :obj:`datasets.Dataset`, columns not
                 accepted by the ``model.forward()`` method are automatically removed.
         """
         if eval_dataset is None and self.eval_dataset is None:
             raise ValueError("Trainer: evaluation requires an eval_dataset.")
-        elif eval_dataset is not None and is_nlp_available() and isinstance(eval_dataset, nlp.Dataset):
+        elif eval_dataset is not None and is_datasets_available() and isinstance(eval_dataset, datasets.Dataset):
             self._remove_unused_columns(eval_dataset, description="evaluation")
         eval_dataset = eval_dataset if eval_dataset is not None else self.eval_dataset
         eval_sampler = self._get_eval_sampler(eval_dataset)
@@ -393,10 +393,10 @@ class Trainer:
 
         Args:
             eval_dataset (:obj:`torch.utils.data.dataset.Dataset`, `optional`):
-                The test dataset to use. If it is an :obj:`nlp.Dataset`, columns not accepted by the
+                The test dataset to use. If it is an :obj:`datasets.Dataset`, columns not accepted by the
                 ``model.forward()`` method are automatically removed.
         """
-        if is_nlp_available() and isinstance(test_dataset, nlp.Dataset):
+        if is_datasets_available() and isinstance(test_dataset, datasets.Dataset):
             self._remove_unused_columns(test_dataset, description="test")
         test_sampler = self._get_eval_sampler(test_dataset)
 
@@ -1200,7 +1200,7 @@ class Trainer:
 
         Args:
             eval_dataset (:obj:`Dataset`, `optional`):
-                Pass a dataset if you wish to override :obj:`self.eval_dataset`. If it is an :obj:`nlp.Dataset`,
+                Pass a dataset if you wish to override :obj:`self.eval_dataset`. If it is an :obj:`datasets.Dataset`,
                 columns not accepted by the ``model.forward()`` method are automatically removed.
 
         Returns:
@@ -1227,7 +1227,7 @@ class Trainer:
 
         Args:
             test_dataset (:obj:`Dataset`):
-                Dataset to run the predictions on. If it is an :obj:`nlp.Dataset`, columns not accepted by the
+                Dataset to run the predictions on. If it is an :obj:`datasets.Dataset`, columns not accepted by the
                 ``model.forward()`` method are automatically removed.
 
         Returns:

--- a/tests/test_trainer.py
+++ b/tests/test_trainer.py
@@ -1,6 +1,6 @@
 import unittest
 
-import nlp
+import datasets
 import numpy as np
 
 from transformers import AutoTokenizer, TrainingArguments, is_torch_available
@@ -200,11 +200,11 @@ class TrainerIntegrationTest(unittest.TestCase):
         x = trainer.eval_dataset.x
         self.assertTrue(np.allclose(preds, 1.5 * x + 2.5))
 
-    def test_trainer_with_nlp(self):
+    def test_trainer_with_datasets(self):
         np.random.seed(42)
         x = np.random.normal(size=(64,)).astype(np.float32)
         y = 2.0 * x + 3.0 + np.random.normal(scale=0.1, size=(64,))
-        train_dataset = nlp.Dataset.from_dict({"input_x": x, "label": y})
+        train_dataset = datasets.Dataset.from_dict({"input_x": x, "label": y})
 
         # Base training. Should have the same results as test_reproducible_training
         model = RegressionModel()
@@ -222,7 +222,7 @@ class TrainerIntegrationTest(unittest.TestCase):
 
         # Adding one column not used by the model should have no impact
         z = np.random.normal(size=(64,)).astype(np.float32)
-        train_dataset = nlp.Dataset.from_dict({"input_x": x, "label": y, "extra": z})
+        train_dataset = datasets.Dataset.from_dict({"input_x": x, "label": y, "extra": z})
         model = RegressionModel()
         trainer = Trainer(model, args, train_dataset=train_dataset)
         trainer.train()

